src/main/python/systemml/mllearn/keras2caffe.py

```diff
@@ -0,0 +1,489 @@
+                                  in_place=True)
+            variance = np.array(blobs[-1])
+            mean = np.array(blobs[-2])
+
+            config = layer.get_config()
+            #Set mean variance and gamma into respective params
+            param = dict()
+            if config['scale']:
+                gamma = np.array(blobs[-3])
+            else:
+                gamma = np.ones(mean.shape, dtype=np.float32)
+
+            if config['center']:
+                beta = np.array(blobs[0])
+                param['bias_term'] = True
+            else:
+                beta = np.zeros(mean.shape, dtype=np.float32)
+                param['bias_term'] = False
+
+            net_params[name] = (mean, variance, np.array(1.0))
+
+            name_scale = name+'_scale'
+            #Scale after batchNorm
+            n[name_scale] = L.Scale(n[name], in_place=True, scale_param=param)
+            net_params[name_scale] = (gamma, beta)
+
+        elif type(layer) == keras.layers.Conv1D:
+            name = layer.name
+            in_names[0]= getInboundLayers(layer)
+            n[name] = L.Convolution()
+
+        elif type(layer) == keras.layers.Conv2D:
+            name = layer.name
+            in_names= getInboundLayers(layer)
+            #Stride
+            if layer.strides is None:
+                stride = (1,1)
+            else:
+                stride= layer.strides
+            #Padding
+            padding = [0,0] #If padding is valid(aka no padding)
+            if layer.padding == 'same': #Calculate the padding for 'same'
+                padding = [layer.kernel_size[0]/2 ,layer.kernel_size[1]/2]
+            #TODO The rest of the arguements including bias, regulizers, dilation,
+            config = layer.get_config()
+            #get bias parameter
+            use_bias = config['use_bias']
+            param = dict(bias_term=use_bias)
+            n[name] = L.Convolution(n[in_names[0].name],kernel_h=layer.kernel_size[0],
+                                    kernel_w=layer.kernel_size[1],stride_h=stride[0],
+                                    stride_w=stride[-1], num_output=layer.filters, pad_h=padding[0], pad_w=padding[1],
+                                    convolution_param=param)
+            weights = blobs
+            blobs[0] = np.array(blobs[0]).transpose((3,2,0,1))
+            print(type(weights))
+            net_params[name] = blobs
+            if layer.activation is not None and layer.activation.__name__ != 'linear':
+                name_act = name + "_activation_" + layer.activation.__name__ #get function string
+                n[name_act] = getActivation(layer,n[name])
+
+        elif type(layer) == keras.layers.MaxPooling2D or type(layer) == keras.layers.AveragePooling2D:
+            name = layer.name
+            in_names= getInboundLayers(layer)
+            if type(layer) == keras.layers.MaxPooling2D:
+                pool = P.Pooling.MAX
+            else:
+                pool = P.Pooling.AVE
```

Comment:
Can you make a note that we need to add average pooling to Caffe2DML and the `nn` library.